### PR TITLE
[DEVOPS-56]  Tierisation

### PR DESCRIPTION
**Pull request by deepfire**
Deployment has been tested according to this instruction: https://issues.serokell.io/issue/DEVOPS-56#comment=96-5541

# Workflow highlights
_..does not include typical boilerplate_

1. edit `cluster.yaml` (or whatever similarly structured config file)
2. `iohk-ops gen-cluster <FILE>` -- emits `cluster.nix`
3. `iohk-ops -c config.yaml deploy`

# TODO
_(nice-to-have leftovers)_

- [ ] a simplified `cluster.yaml` for small-fast-devvy deployments

### CODE REVIEW

**Comment by deepfire** [comment]
@domenkozar, I need to figure out what values to pass to `--static-peer` (note the singular) and `--kademlia-peer`, but otherwise it is complete -- modulo the not-yet-implemented `cardano-node` option semantics it depends on.

File: modules/cardano-node-prod.nix [hunk 1]
```diff
@@ -12,10 +12,10 @@ with (import ./../lib.nix);
   # Initial block is big enough to hold 3 months of transactions
   deployment.ec2.ebsInitialRootDiskSize = mkForce 700;
 
-  deployment.ec2.elasticIPv4 = resources.elasticIPs.${"nodeip" + toString config.services.cardano-node.testIndex};
+  deployment.ec2.elasticIPv4 = resources.elasticIPs.${"nodeip-" + config.services.cardano-node.nodeIndex};
```
Comment (jmitchell) [inline modules/cardano-node-prod.nix]: `toString` is still necessary, isn't it?

Comment (deepfire) [inline modules/cardano-node-prod.nix]: Yes! Thank you!

File: default.nix [hunk 2]
```diff
@@ -53,17 +57,9 @@ let
...
-  iohk-ops-extra-runtime-deps = [
-    pkgs.git pkgs.nix-prefetch-scripts
-  ];
-  cabal2nixpkgs = rec {
-    # extra packages to expose, that have no relation to pkgs/default.nix
-    stack2nix = compiler.callPackage ./pkgs/stack2nix.nix {};
+    iohk-ops-extra-runtime-deps = [
+      pkgs.git pkgs.nix-prefetch-scripts self.yaml
+    ];
     iohk-ops = pkgs.haskell.lib.overrideCabal
```
Comment (cleverca22) [inline default.nix]: with this change, `iohk-ops` now depends on the versions cardano's stack.yaml picked, and updating `pkgs/default.nix` will invalidate the cached build of `iohk-ops`

**Pull request closed without merging**